### PR TITLE
wasmer: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **wasmer/2.0.0**

wasmer is one of WebAssembly Runtimes.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/wasmer/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+sources:
+  "2.0.0":
+    Windows:
+      "x86_64":
+        url: "https://github.com/wasmerio/wasmer/releases/download/2.0.0/wasmer-windows-amd64.tar.gz"
```
Comment (madebr) [inline recipes/wasmer/all/conandata.yml:5]: I don't think this contains a static library.
`wasmer.dll` is the binary, and `wasmer.lib` is the import library.
The archive does not contain a stand-alone static library.

Comment (madebr) [reply recipes/wasmer/all/conandata.yml:5]: If needed, you can create an import library using `pexports` (we have a recipe for that.
I would create an issue at the wasmer repo.
I did the same thing for wasmtime and they fixed it in the next release.

File: recipes/wasmer/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        else:
+            self.copy("wasmer.lib", src=srclibdir, dst="lib", keep_path=False)
+            self.copy("libwasmer.a", src=srclibdir, dst="lib", keep_path=False)
+
+        self.copy("LICENSE", dst="licenses", src=self.source_folder)
+
+    def package_info(self):
+        if self.options.shared:
+            if self.settings.os == "Windows":
+                self.cpp_info.libs = ["wasmer.dll"]
```
Comment (madebr) [inline recipes/wasmer/all/conanfile.py:71]: You can't link against a dll. You need an import library for that.

Comment (toge) [reply recipes/wasmer/all/conanfile.py:71]: Sorry for my lack of Windows development experiments.
I will try to resolve it.

Thank your advice. You always help me lot!

**Comment by madebr** [comment]
@toge 
See https://github.com/toge/conan-center-index/pull/1

I also opened https://github.com/wasmerio/wasmer/issues/2680 to get the Windows DLL fixed.

**Comment by toge** [comment]
@madebr
Sorry for late response.
I merged your PR. Thanks a lot!

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/wasmer/all/conandata.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          url: "https://github.com/wasmerio/wasmer/releases/download/2.0.0/wasmer-windows-amd64.tar.gz"
+          sha256: "94d0cd885ace472eb3173f6e47fd05c82c4aa7f665a2d54861d3c4903024beee"
+    Linux:
+      "x86_64":
+        "gcc":
+          url: "https://github.com/wasmerio/wasmer/releases/download/2.0.0/wasmer-linux-amd64.tar.gz"
+          sha256: "b56bb7edc456a7b6f22e35d9b9926d8b5ed806d9081197b8a28d586ef96c1579"
+    Macos:
+      "x86_64":
+        "gcc":
```
Comment (prince-chrismc) [inline recipes/wasmer/all/conandata.yml:15]: macos gcc :confused:

Comment (prince-chrismc) [reply recipes/wasmer/all/conandata.yml:15]: I found compiler alias

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**